qiskit_experiments/library/characterization/analysis/stark_ramsey_fast_analysis.py

```diff
@@ -0,0 +1,303 @@
+        To precisely fit the results, this analysis uses different model parameters
+        for positive (:math:`x > 0`) and negative (:math:`x < 0`) shift domains.
+
+        To obtain the initial guess, the following calculation is employed in this analysis.
+        First, oscillations in each quadrature are normalized and differentiated.
+
+        .. math ::
+
+            \dot{F}_X = \frac{\partial}{\partial x} \bar{F}_X = dt \frac{d}{dx} f_S \bar{F}_Y, \\
+            \dot{F}_Y = \frac{\partial}{\partial x} \bar{F}_Y = - dt \frac{d}{dx} f_S \bar{F}_X. \\
+
+        The root sum square of above quantities yields
+
+        .. math ::
+
+            \sqrt{\dot{F}_X + \dot{F}_Y} = dt \frac{d}{dx} f_S = dt (c_1 + 2 c_2 x + 3 c_3 x^2).
+
+        By using this synthesized data, one can estimate the initial guess of the
+        polynomial coefficients by quadratic regression.
+        This fit protocol is independently conducted for the experiment data on the
+        positive and negative shift domain.
+
+    # section: fit_parameters
+
+        defpar \rm amp:
+            desc: Amplitude of both series.
+            init_guess: Median of root sum square of Ramsey X and Y oscillation.
+            bounds: [0, 1]
+
+        defpar \rm offset:
+            desc: Base line of all series.
+            init_guess: Roughly the average of the data.
+            bounds: [-1, 1]
+
+        defpar dt:
+            desc: Fixed parameter of :math:`2 \pi t_S`, where :math:`t_S` is
+                the ``stark_length`` experiment option.
+            init_guess: Automatically set from metadata when this analysis is run.
+            bounds: None
+
+        defpar c_1^+:
+            desc: The linear term coefficient of the positive Stark shift.
+            init_guess: See the fit model description.
+            bounds: None
+
+        defpar c_2^+:
+            desc: The quadratic term coefficient of the positive Stark shift.
+            init_guess: See the fit model description.
+            bounds: None
+
+        defpar c_3^+:
+            desc: The cubic term coefficient of the positive Stark shift.
+            init_guess: See the fit model description.
+            bounds: None
+
+        defpar c_1^-:
+            desc: The linear term coefficient of the negative Stark shift.
+            init_guess: See the fit model description.
+            bounds: None
+
+        defpar c_2^-:
+            desc: The quadratic term coefficient of the negative Stark shift.
+            init_guess: See the fit model description.
+            bounds: None
+
+        defpar c_3^-:
+            desc: The cubic term coefficient of the negative Stark shift.
+            init_guess: See the fit model description.
+            bounds: None
+
+        defpar f_\epsilon:
+            desc: Constant phase accumulation which is independent of the Stark tone amplitude.
+            init_guess: 0
+            bounds: None
```

Comment:
Should these match the model? I am wondering about `c_3^-` vs `c3_neg` and `f_\epsilon` vs `f_err`.

Reply:
Okay, hmm, I see how in the docstring a TeX format works better, but what I wonder about is if the entries here are documenting the parameters in the docstring discussion or the parameters that are used in the lmfit model. Perhaps the TeX names are best, but the descriptions should mention the lmfit names as well for reference?

Reply:
Good idea. Done in [76f1fb5](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1009/commits/76f1fb576ed04211dfa67f5a7c6339f52ca735da)

Reply:
Nice! One very minor question -- should we reference LMFIT directly or just fit parameters? I was mainly interested in the documentation matching up with the fit parameters since they are user facing if the user uses `return_fit_parameters` and gets the `CurveFitResult`. I don't know if LMFIT matters for that or is an implementation detail?

Reply:
Replaced with model parameter name in [e751212](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1009/commits/e7512120fda30f13c159c2ad6e14b0399dddc62f)

Reply:
I think I misunderstood your intention. The model name is further replaced with lengthy name in the fit result object because something like `c3_neg` doesn't really meaningful without context of this experiment, and indeed we can search for the name in the result database. This is updated to the name in the database in [9415e1d](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1009/commits/9415e1df3a6fbd55287446aeeb4810bfd49a797b)

Reply:
Actually, I did mean the `c3_neg` model names, but documenting the longer names is good as well and probably even better, though we should think about whether we want a consistent way of documenting tying all three together -- documentation symbols, model parameter name, and analysis result name (when it is directly a model parameter).

The case I was thinking of was the user using the `return_fit_parameters` analysis option which returns an object with all of the model parameters. I was thinking it would be nice to document those since otherwise the user needs to read the code to check what those names correspond. It is also valuable to document the analysis result names.